tests/tests/test-xcm/test-xcmv3-new-instructions.ts

```diff
@@ -0,0 +1,382 @@
+        .withdraw_asset()
+        .buy_execution()
+        .query_pallet(1002)
+        .as_v3();
+
+      // Mock the reception of the xcm message
+      await injectHrmpMessage(context, paraId, {
+        type: "XcmVersionedXcm",
+        payload: xcmMessage,
+      } as RawXcmMessage);
+      await context.createBlock();
+
+      // Search for Unroutable error
+      const records = (await context.polkadotApi.query.system.events()) as any;
+      const events = records.filter(
+        ({ event }) => event.section == "xcmpQueue" && event.method == "Fail"
+      );
+      expect(events).to.have.lengthOf(1);
+      expect(events[0].toHuman().event.data.error).equals("Unroutable");
+    });
+
+    it("Should execute ExpectPallet (NameMismatch)", async function () {
+      const xcmMessage = new XcmFragment(dotAsset)
+        .withdraw_asset()
+        .buy_execution()
+        .expect_pallet()
+        .as_v3();
+
+      // Mock the reception of the xcm message
+      await injectHrmpMessage(context, paraId, {
+        type: "XcmVersionedXcm",
+        payload: xcmMessage,
+      } as RawXcmMessage);
+      await context.createBlock();
+
+      // Search for NameMismatch error
+      const records = (await context.polkadotApi.query.system.events()) as any;
+      const events = records.filter(
+        ({ event }) => event.section == "xcmpQueue" && event.method == "Fail"
+      );
+      expect(events).to.have.lengthOf(1);
+      expect(events[0].toHuman().event.data.error).equals("NameMismatch");
+    });
+
+    it("Should execute ReportTransactStatus (Transport error)", async function () {
+      const xcmMessage = new XcmFragment(dotAsset)
+        .withdraw_asset()
+        .buy_execution()
+        .report_transact_status(1000)
+        .as_v3();
+
+      // Mock the reception of the xcm message
+      await injectHrmpMessage(context, paraId, {
+        type: "XcmVersionedXcm",
+        payload: xcmMessage,
+      } as RawXcmMessage);
+      await context.createBlock();
+
+      // Search for Transport error
+      const records = (await context.polkadotApi.query.system.events()) as any;
+      const events = records.filter(
+        ({ event }) => event.section == "xcmpQueue" && event.method == "Fail"
+      );
+      expect(events).to.have.lengthOf(1);
+      expect(events[0].toHuman().event.data.error).equals("Transport");
+    });
+
+    it("Should execute UnpaidExecution (BadOrigin)", async function () {
+      const xcmMessage = new XcmFragment(dotAsset)
+        .withdraw_asset()
+        .buy_execution()
+        .unpaid_execution(1)
+        .as_v3();
+
+      // Mock the reception of the xcm message
+      await injectHrmpMessage(context, paraId, {
+        type: "XcmVersionedXcm",
+        payload: xcmMessage,
+      } as RawXcmMessage);
+      await context.createBlock();
+
+      // Search for BadOrigin error
+      const records = (await context.polkadotApi.query.system.events()) as any;
+      const events = records.filter(
```

Comment:
same comment as in the `tests/tests/test-xcm/test-xcmv3-max-weight-instructions.ts`